### PR TITLE
internal/pubsub/rabbitpubsub: succeed on open of non-existent resource

### PR DESCRIPTION
**Pull request by jba**
OpenTopic and OpenSubscription now succeed even if the resource doesn't exist.
They no longer need to return an error.

Also, implement the test harness MakeNonexitentSubscription method.

### CODE REVIEW

**Comment by ijt** [comment]
I'm guessing this fixes #891.

File: internal/pubsub/rabbitpubsub/rabbit.go [hunk 1]
```diff
@@ -228,18 +211,12 @@ type subscription struct {
...
-	s := &subscription{
+func newSubscription(conn *amqp.Connection, name string) *subscription {
+	return &subscription{
 		conn:     conn,
 		queue:    name,
 		consumer: fmt.Sprintf("c%d", atomic.AddInt64(&nextConsumer, 1)),
 	}
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	if err := s.establishChannel(); err != nil {
```
Comment (ijt) [inline internal/pubsub/rabbitpubsub/rabbit.go]: Doesn't this call have to be done somewhere else now? I don't see it.

Comment (jba) [reply internal/pubsub/rabbitpubsub/rabbit.go]: It's in ReceiveBatch and SendAcks.

**Comment by jba** [comment]
I'll add the Fixes line before I submit.

**Comment by jba** [comment]
Awaiting @ijt LGTM.

**Comment by jba** [comment]
@ijt PTAL

File: internal/pubsub/rabbitpubsub/rabbit.go (hunk 1 above)
Comment (ijt) [reply internal/pubsub/rabbitpubsub/rabbit.go]: Ah, right.

**Pull request merged**